QuestionAnswering/QA_ICL/README.md

```diff
@@ -1,15 +1,199 @@
+        ...
+      }
+      ```
+   - Update backend entpoint configuration in NextApp `.env` file to the redirection address
+   - Rebuild Image for `NextApp` and launch the container
+   - Troubleshoot
+     - CORS error when submitting question: 
+       - This is caused by incorrect setting of the backend redirection
+
+
+### Development Setup
+1. Clone the TWA repository and go to the QA_ICL folder
+
+    ```
+    cd TheWorldAvatar/QuestionAnswering/QA_ICL/
+    ```
+
+2. Setup the backend: 
+   - This is a summary of the backend setup instructions. More details can be found in [backend/README.md](./backend/README.md)
+   - Data Preparation
+      - Follow the instructions in `backend/README.md` to prepare required data and configure parameters.
+   - Triton Setup
+      - Follow instructions [here](https://github.com/cambridge-cares/TheWorldAvatar/blob/main/QuestionAnswering/QA_ICL/backend/triton_inference_server/README.md#development)
+   - FastApi Setup
+      - Create a virtual environment (recommended):
+        ```
+        conda create --name qa_backend python==3.10
+        conda activate qa_backend
+        ```
+        Install relavent dependencies
+        ```
+        pip install -r requirements.txt
+        ```
+      - Provide credentials as stated [here](https://github.com/cambridge-cares/TheWorldAvatar/tree/main/QuestionAnswering/QA_ICL/backend/fastapi_app#configurable-parameters)
+      - Ingest data to `Redis`
+        ```
+        python -m ingest_data --redis_host localhost --text_embedding_backend triton --text_embedding_url localhost:8001 --drop_index --invalidate_cache
+        ```
+      - Start server as stated [here](https://github.com/cambridge-cares/TheWorldAvatar/tree/main/QuestionAnswering/QA_ICL/backend/fastapi_app#steps)
+
+3. Set up the frontend:
+   - Navigate to the `frontend/next_app_marie` directory.
+   - Local Install:
+      ```
+      npm install
+      ```
+    - Configure the backend endpoint in the `.env` file.
+    - More information can be found at [here](https://github.com/cambridge-cares/TheWorldAvatar/tree/main/QuestionAnswering/QA_ICL/frontend/next_app_marie#development)
+4. Set up the data generation environment:
+    - Create a virtual environment (recommended):
+      ```
+      conda create --name data-env python=3.8
+      conda activate data-env
+      ``` 
+
+    - Install dependecies:
+      ```
+      cd data_generation
+      pip install -r requirements.txt
+      ```
+
+
+### API Documentation
+Once the backend is running, you can access the API documentation at: `http://localhost:5000/docs`.
+
+This interactive documentation provides detailed information about available endpoints, request/response formats, and allows you to test the API directly from your browser.
+
+### Usage
+After completing the deployment setup:
+1. Ensure the backend is running (should be active from step 2 of deployment)
+2. Ensure the frontend is running (should be active from step 3 of deployment)
```

Comment:
Wondering if the 1. and 2. bullet points can be omitted. If you wanna keep these points, what are the specific actions to test that the backend/frontend is running? Actually these points should be part of the deployment steps as well...